Assignment submission for review

Carvas91 opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

I'm adding code to the shell script that was empty in the first place.

I had to use a different command (curl instead of wget) to download the zip file since the one in the original script was downloading an empty file.

I corrected the file paths in the comments in the shell script (./data/raw/rawdata)

I removed an invalid command "rf -rf ./data". This was introduced after pulling and merging the coworkers branch.

Update of the readme file

## What did you learn from the changes you have made?

I learned how to use wildcards. It reminded me a little bit about SQL.

I learned that I should check for the coworker's code. It might be incorrect.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

Yes, I thought about testing the script command by command instead of running the whole script all at once. I did this because I wanted to be super sure about the output.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

Yes:
The wget command didn't work for me. It'd download an empty zip file. I had to use a curl command instead.
I noticed an incorrect command, rf -rf ./data after merging with the coworker branch. I read the requirement file and noticed that that command wasn't required, so I deleted it.

## How were these changes tested?

I ran the script command by command and then I created a new clean environment for the script to run all at once and it also worked.

## A reference to a related issue in your repository (if applicable)

The pull request was new for me. So I'm getting used to it.

## Checklist
- [x] I can confirm that my changes are working as intended

dtxe committed:
delete ip logs... WARNING UNTESTED!

dtxe committed:
initialize README file with company name

Carvas91 committed:
first commit, mkdir data

Carvas91 committed:
script works perfectly after testing

Carvas91 committed:
Merge branch 'coworker-changes' of https://github.com/UofT-DSI/shell into assignment

Carvas91 committed:
corrected script after merging with coworker's

Carvas91 committed:
final test of the script; it worked

Carvas91 commented:
I think steps 5 and 6 from the autogrades don't match what's actually required in the assignment document.
The assignment document asks for "# 2. Move the ./rawdata directory to ./data/raw" so, in data/raw there's going to be the rawdata dir, and not the server log files. The files are actually inside the rawdata dir.
Kindly confirm this.
Thank you,
Carlos

monzchan commented:
@Carvas91 Great job! 

In question 2, you are moving the rawdata directory into data/raw directory by "mv rawdata data/raw".
--> If data/raw exists, only the contents of rawdata are moved into it.
--> If data/raw does not exist, the entire rawdata directory is moved and renamed to raw.
--> this command should not have moved the rawdata dir into data/raw dir as a subdirectory. (you may try mv rawdata data/raw/ --> this will move the rawdata dir into data/raw dir as a subdirectory)

Carvas91 committed:
Shell assignment updated after Monz Chan and Anjali feedback.

Carvas91 commented:
Dear @monzchan 
Thank you very much for your feedback. I've gone and updated the code, however I honestly think the requirements from the assignment don't match the autograder.

In question # 2. Move the ./rawdata directory to ./data/raw. It actually asks us to move the rwdata directory(folder) and not just its content. However, I've modified the code so that only the content of the rawdata directory moves to the raw directory. This is the code I used: `mv ./rawdata/* ./data/raw`.

After applying this update, the autograder still isn't giving me full marks because "rawdata not moved to data/raw". I think the autograder is looking for that directory inside data/raw but it won't find it because I'm just moving the rawdata directory content.

Could you please confirm that?

monzchan commented:
@Carvas91 
There's another participant experience similar question as you do, and here is the explanation:
Command - 1: 
  mkdir ./data/raw
   mv rawdata ./data/raw
Problem - No server log files in data/raw
These commands create the raw dir, and move rawdata dir into the raw dir, so the content of rawdata dir is not in data/raw

Command - 2:
   mkdir ./data/raw
   mv rawdata/ ./data/raw*
Problem - rawdata not moved to data/raw
These commands create the raw dir, and move only the content (files & subdirectories) of rawdata dir into the raw dir. But the original raw data directory will still exist.

Command - 3:
Without creating a ./data/raw dir, you would be able to move the ./rawdata directory to ./data/raw
   mv ./rawdata ./data/raw
This command moves the entire rawdata dir (inc. all files, subdirectories, and metadata) into ./data and renames it to raw. After execution, the rawdata directory no longer exists in its original location.

I guess you may find their difference and know which commands would be the best for answering question 2. Let me know if there is any question.

Carvas91 committed:
Third iteration after Moniz feedback on th PR.

monzchan commented:
@Carvas91 Excellent! Your effort is highly appreciated!

monzchan reviewed (approved):
Wonderful job!

Carvas91 commented:
@monzchan Thank you very much for your guidance!
